src/server/internalServer.cpp

```diff
@@ -336,7 +338,7 @@ InternalServer::get_matching_if_none_match_etag(const RequestContext& r) const
 
 std::unique_ptr<Response> InternalServer::build_homepage(const RequestContext& request)
 {
-  return ContentResponse::build(*this, RESOURCE::templates::index_html, get_default_data(), "text/html; charset=utf-8", true);
+  return ContentResponse::build(*this, !m_indexTemplateString.empty() ? m_indexTemplateString : RESOURCE::templates::index_html, get_default_data(), "text/html; charset=utf-8", true);
```

Comment:
This could be made once at `InternalServer` construction.
No need to check every time if m_indexTemplateString is empty as it will never change.